cot/src/form/fields/select.rs

```diff
@@ -0,0 +1,750 @@
+    /// The list of available choices for the select field.
+    /// If not set, the default choices from [`SelectChoice::default_choices`]
+    /// will be used.
+    pub choices: Option<Vec<T>>,
+    /// Custom text for the empty option when the field is not required.
+    /// If not set, "—" will be used as the default empty option text.
+    /// If the field is required, no empty option will be displayed, unless
+    /// this is set explicitly.
+    pub none_option: Option<String>,
+}
+
+impl<T> Default for SelectFieldOptions<T> {
+    fn default() -> Self {
+        Self {
+            choices: None,
+            none_option: None,
+        }
+    }
+}
+
+impl<T: SelectChoice + Send> Display for SelectField<T> {
+    fn fmt(&self, f: &mut Formatter<'_>) -> std::fmt::Result {
+        const DEFAULT_NONE_OPTION: &str = "—";
+
+        let value = if let Some(value) = self.value.clone() {
+            IndexSet::from([value])
+        } else {
+            IndexSet::new()
+        };
+
+        let none_option = if let Some(none_option) = &self.custom_options.none_option {
+            Some(none_option.as_str())
+        } else if self.options.required {
+            None
+        } else {
+            Some(DEFAULT_NONE_OPTION)
+        };
+        render_select(
+            f,
+            self,
+            false,
+            none_option,
+            None,
+            self.custom_options.choices.as_ref(),
+            &value,
+        )
+    }
+}
+
+impl<T: SelectChoice + Send> HtmlSafe for SelectField<T> {}
+
+/// A form field for a multiple-choice select box.
+///
+/// This field allows users to select multiple values from a predefined list of
+/// choices. Unlike [`SelectField`], this field can accept multiple selections
+/// and renders as a multi-select HTML element.
+#[derive(Debug)]
+pub struct SelectMultipleField<T> {
+    options: FormFieldOptions,
+    custom_options: SelectMultipleFieldOptions<T>,
+    value: IndexSet<String>,
+}
+
+impl<T> SelectMultipleField<T> {
+    /// Returns an iterator over the selected values as string slices.
+    pub fn values(&self) -> impl Iterator<Item = &str> {
+        self.value.iter().map(AsRef::as_ref)
+    }
+}
+
+impl<T: SelectChoice + Send> FormField for SelectMultipleField<T> {
+    type CustomOptions = SelectMultipleFieldOptions<T>;
+
+    fn with_options(options: FormFieldOptions, custom_options: Self::CustomOptions) -> Self {
+        Self {
+            options,
+            custom_options,
+            value: IndexSet::new(),
+        }
+    }
+
+    fn options(&self) -> &FormFieldOptions {
+        &self.options
+    }
+
+    fn value(&self) -> Option<&str> {
+        None
+    }
+
+    async fn set_value(&mut self, field: FormFieldValue<'_>) -> Result<(), FormFieldValueError> {
+        self.value.insert(field.into_text().await?);
+        Ok(())
+    }
+}
+
+/// Custom options for a [`SelectMultipleField`].
+#[derive(Debug, Clone)]
+pub struct SelectMultipleFieldOptions<T> {
```

Comment:
Why did you not include an empty option here?

Reply:
Because it's not really needed. For a multiple-selection field there's already an empty option: it's just empty selection. For a regular dropdown, the web browsers select the first option by default, which might be not what the user wants, hence an explicit `None` option.